### ISSUE TITLE
Placeholder attribute on control instead of @placeholder argument on element does not render placeholder

### ISSUE THREAD

**Issue by dagroe** [kaliber5/ember-bootstrap-power-select#99]
While using `@placeholder` on a `Form.element` of type power-select works, it is deprecated. The alternative of using the `placeholder` attribute on the `Form.element.control` doesn't work (for me) however.

This renders the palceholder correctly but gives a deprecation warning:

```hbs
<Form.element
    @controlType="power-select"
    @property="property"
    @onChange={{this.setProperty}}
    @options={{this.propertyChoices}}
    @label="some label"
    @placeholder="placeholder"
    as |Element|
    >
    <Element.control
      as |option|
    >
      {{option}}
    </Element.control>
  </Form.element>
```

This doesn't render the placeholder:
```hbs
<Form.element
    @controlType="power-select"
    @property="property"
    @onChange={{this.setProperty}}
    @options={{this.propertyChoices}}
    @label="some label"
    as |Element|
    >
    <Element.control
      placeholder="placeholder"
      as |option|
    >
      {{option}}
    </Element.control>
  </Form.element>
```

Is this an issue or am I making a mistake?

**Comment by jelhan** [comment]
If using `@controlType="power-select"` through `ember-bootstrap-power-select` the yielded `<Element.control>` is a `<PowerSelect>` component. That one accept a `@placeholder` argument. It forwards HTML attributes to the underlying `<BasicDropdown>`. That's not what you want.

TL;DR: Your code should look like this:

```hbs
  <Form.element
    @controlType="power-select"
    as |Element|
  >
    <Element.control
      @placeholder="placeholder"
      as |option|
    >
      {{option}}
    </Element.control>
  </Form.element>
```

**Comment by dagroe** [comment]
Ah, I see. That works. Thanks a lot!

**Comment by tnp** [comment]
It looks like 

addon/components/bs-form/element/control/power-select.hbs

was recently changed to remove the line

`@placeholder={{@placeholder}}`

which means that the placeholder option is not passed to the placeholder component, so the placeholder does not appear.

Restoring that line will fix the problem.

**Comment by jelhan** [comment]
It was removed in #113. @simonihmig Can you explain why? Or was it just an accident?

**Comment by mwasilewski** [comment]
any news on that?

**Comment by zinyando** [comment]
I'm also wondering the same, seems now we are "forced" to use a `placeholderComponent`? It would be good to get an official position on this.

**Comment by jelhan** [comment]
I assume it was a mistake. I would accept a merge request adding `@placeholder` argument back to `<BsForm::Element::Control::PowerSelect>` and `<BsForm::Element::Control::PowerSelectMultiple>`. It should only add test coverage. The merge request should include test coverage.

I think most of the code can be pulled in again from #113. But tests need to be updated to not rely on `<BsForm::Element>` passing `@placeholder` to the control. Instead the argument should be set on yielded control directly. See https://github.com/kaliber5/ember-bootstrap-power-select/issues/99#issuecomment-670380846.

Sorry that it took so long for an official statement.

**Comment by simonihmig** [comment]
Sorry for the late feedback, this slipped through unfortunately.

> It was removed in #113. @simonihmig Can you explain why? Or was it just an accident?

Looking back, I *guess* my thought was that as the other built-in control components also don't accept a `@placeholder` *argument*  anymore (they do accept a `placeholder` attribute though of course), this needed to be applied also to power-selects. But I guess this was wrong, when power-selects do still expect users to use a `@placeholder` argument. I think the same applies to `@disabled` also being removed as part of #113.

So yes, reverting these parts of #113 seems right. Stupid me, sorry! 😔

I someone is able to contribute a PR, bringing back the arguments and the removed test cases, that would be awesome! Otherwise I hope I can do that next weekend...
